core/domain/question_services.py

```diff
@@ -712,3 +714,61 @@ def get_interaction_id_for_question(question_id):
     if question is None:
         raise Exception('No questions exists with the given question id.')
     return question.question_state_data.interaction.id
+
+
+def untag_deleted_misconceptions(committer_id, skill_id, skill_description):
+    """Untags deleted misconceptions from questions belonging
+    to a skill with the provided skill_id.
+
+    Args:
+        committer_id: str. The id of the user who triggered the update.
+        skill_id: str. The skill id.
+        skill_description: str. The description of the skill.
+    """
+    question_skill_links = get_question_skill_links_of_skill(
+        skill_id, skill_description)
+    question_ids = [model.question_id for model in question_skill_links]
+    questions = question_fetchers.get_questions_by_ids(question_ids)
+    skill = skill_fetchers.get_skill_by_id(skill_id)
+    skill_misconception_ids = (
+        [
+            skill.generate_skill_misconception_id(misconception.id)
+            for misconception in skill.misconceptions
+        ]
+    )
+    for question in questions:
+        change_list = []
+        inapplicable_skill_misconception_ids = (
+            question.inapplicable_skill_misconception_ids)
+        deleted_inapplicable_skill_misconception_ids = (
+            utils.compute_list_difference(
+                inapplicable_skill_misconception_ids, skill_misconception_ids))
+        for misconception_id in deleted_inapplicable_skill_misconception_ids:
+            old_misconception_ids = list(
+                question.inapplicable_skill_misconception_ids)
+            question.inapplicable_skill_misconception_ids.remove(
+                misconception_id)
+            change_list.append(question_domain.QuestionChange({
```

Comment:
Why are we generating `n` commands? is there any issue with having one command considering the command name is `update_question_property`?